examples/30_extended/fetch_runtimes.py

```diff
@@ -0,0 +1,132 @@
+                )
+            )
+        print()
+
+
+############################################################################
+# Case 1: Running a Random Forest model on an OpenML task
+# *******************************************************
+# We'll run a Random Forest model and obtain an OpenML run-object. We can
+# see the evaluations recorded per fold for the dataset and the information
+# available for this run.
+
+run1 = openml.runs.run_model_on_task(model=clf, task=task)
+measures = run1.fold_evaluations
+
+# The timing and performance metrics available are
+for key in measures.keys():
+    print(key)
+
+# The performance metric is recorded under `predictive_accuracy` per fold
+# and can be retrieved as:
+for repeat, val1 in measures["predictive_accuracy"].items():
+    for fold, val2 in val1.items():
+        print("Repeat #{}-Fold#{}: {}".format(repeat, fold, val2))
+    print()
+
+# The remaining entries recorded in `measures` are the runtime records
+# related as:
+# usercpu_time_millis = usercpu_time_millis_training + usercpu_time_millis_testing
+# wall_clock_time_millis = wall_clock_time_millis_training + wall_clock_time_millis_testing
+#
+# The timing measures recorded as `*_millis_training` contain the per
+# repeat-fold timing incurred for the executing of the `.fit()` procedure
+# of the model. For `usercpu_time_*` the time recorded using `time.process_time()`
+# is converted to milliseconds and stored. Similarly, `time.time()` is used
+# to record the time entry for `wall_clock_time_*`. *_millis_testing`
+# follows the same procedure but for time taken for the `.predict()` procedure.
+
+# Comparing the CPU and wall-clock training times of the Random Forest model
+print_compare_runtimes(measures)
+
+############################################################################
+# Case 2: Running a Random Forest model on an OpenML task in parallel
+# *******************************************************************
+# Running the same model on the same task but with parallelism (n_jobs > 1)
+
+run2 = openml.runs.run_model_on_task(model=clf, task=task, n_jobs=4)
+measures = run2.fold_evaluations
+
+# Comparing the CPU and wall-clock training times of the Random Forest model
+print_compare_runtimes(measures)
+
+############################################################################
+# Case 3: Running and benchmarking HPO algorithms with their runtimes
+# *******************************************************************
+# We shall now optimize the same RandomForest model for the same task using
+# scikit-learn HPO support by using GridSearchCV to optimize our earlier
+# RandomForest model's parameter `n_estimators`. sklearn also provides a
+# `refit_time_` for such HPO models, i.e., the time incurred by training
+# and evaluating the model on the best found parameter setting. OpenML
+# includes this in the `wall_clock_time_millis_training` measure recorded.
```

Comment:
Could you please also show how to see the fit times stored in the trajectory and the refit time?

Reply:
I had to commit again to remove the inclusion of `refit_time` in the `user_defined_measures` recorded in a run object under `fold_evaluations` since the OpenML server recognises it as an illegal extra field when trying to publish such run objects. 

In order to bypass this and show an example of how one could extract refit time, we could make the following change,
```
# currently
wall_clock_time_millis = wall_clock_time_millis_training + wall_clock_time_millis_testing
# where
wall_clock_time_millis_training = wall_clock_time_millis_training + refit_time
```
we can change to:
```
wall_clock_time_millis = wall_clock_time_millis_training + refit_time + wall_clock_time_millis_testing
```
Then in the example, we can show (although quite verbose) way to extract refit time as,
```python
refit_time = run.fold_evaluations["wall_clock_time_millis"][repeat][fold] - \
     run.fold_evaluations["wall_clock_time_millis_training"][repeat][fold] - \
     run.fold_evaluations["wall_clock_time_millis_testing"][repeat][fold]
```

Wrt fit times for the trace/trajectory, I did add some things based on my interpretation. Would gladly re-iterate or change based on feedback.

Reply:
Currently, the example doesn't show how to retrieve `refit_time`. Should I make the change as I detailed above or let it be as it is now?

Reply:
I think it would be great if we can add this

Reply:
Done.